lib/ansible/modules/system/interfaces_file.py

```diff
@@ -320,7 +320,7 @@ def addOptionAfterLine(option, value, iface, lines, last_line_dict, iface_option
 def write_changes(module, lines, dest):
 
     tmpfd, tmpfile = tempfile.mkstemp()
-    f = os.fdopen(tmpfd, 'wb')
+    f = os.fdopen(tmpfd, 'w')
     f.writelines(lines)
```

Comment:
It's better to keep the file opened in binary mode and then transform the lines into bytes explicitly.  That gives the program explicit control over encoding on both python2 and python3 rather than leaving it up to the whims of python and the underlying platform's configuration.  So something like this:

``` python
from ansible.module_utils._text import to_bytes

[...]

f = os.fdopen(tmpfd, 'wb')
f.writelines(to_bytes(lines, errors='surrogate_or_strict'))
f.close()
```

Reply:
@abadger 

Thanks fo review.
Applied that.